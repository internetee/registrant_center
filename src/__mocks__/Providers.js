import { CookiesProvider } from 'react-cookie';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import translations from '../translations';
import state from './mockReduxState';

const lang = 'et';

export default function Providers({ children }) {
    return (
        <BrowserRouter>
            <Provider store={state}>
                <CookiesProvider>
                    <IntlProvider
                        key={lang}
                        defaultLocale="et"
                        locale={lang}
                        messages={translations[lang]}
                    >
                        {children}
                    </IntlProvider>
                </CookiesProvider>
            </Provider>
        </BrowserRouter>
    );
}
