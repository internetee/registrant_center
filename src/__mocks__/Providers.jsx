import { CookiesProvider } from 'react-cookie';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router';
import translations from '../translations';
import PropTypes from 'prop-types';

const lang = 'et';

export default function Providers({ children, store }) {
    return (
        <BrowserRouter>
            <Provider store={store}>
                <CookiesProvider>
                    <IntlProvider
                        defaultLocale="et"
                        key={lang}
                        locale={lang}
                        messages={translations[lang]}
                    >
                        {children}
                    </IntlProvider>
                </CookiesProvider>
            </Provider>
        </BrowserRouter>
    );
}

Providers.propTypes = {
    children: PropTypes.node.isRequired,
    store: PropTypes.object.isRequired,
};
