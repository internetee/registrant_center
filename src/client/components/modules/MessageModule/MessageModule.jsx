import React from 'react';
import PropTypes from 'prop-types';
import { Message, Container, Icon } from 'semantic-ui-react';
import staticMessages from '../../../utils/staticMessages.json';

/**
 * @return {null}
 */
function MessageModule({ message, lang, formMessage }) {
  if (message) {
    let text = staticMessages[lang][message.type][message.code];
    // Select message from global messages
    if (!text) {
      text = staticMessages[lang].global[message.code];
    }
    return (
      <Message icon={formMessage} positive={message.code >= 200 || message.code < 300} negative={message.code >= 400} className={ (!formMessage ? 'status-message' : '') }>
        { formMessage ? (
          <React.Fragment>
            <Icon name={(message.code >= 200 || message.code < 300) ? 'check circle' : 'exclamation triangle'} />
            <Message.Content>
              <Message.Header>{ text }</Message.Header>
            </Message.Content>
          </React.Fragment>
        ) : (
          <Container>
            <Message.Header>
              { text }
            </Message.Header>
          </Container>
        )}
      </Message>
    );
  }
  return null;
}

MessageModule.propTypes = {
  formMessage: PropTypes.bool,
};

MessageModule.defaultProps = {
  formMessage: false,
};

export default MessageModule;
