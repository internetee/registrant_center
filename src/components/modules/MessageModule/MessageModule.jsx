import React from 'react';
import { FormattedMessage } from 'react-intl';
import { Message, Container, Icon } from 'semantic-ui-react';
import PropTypes from 'prop-types';

function MessageModule({ message, formMessage }) {
    if (message) {
        return (
            <Message
                className={!formMessage ? 'status-message' : ''}
                icon={formMessage}
                negative={message.code >= 400}
                positive={message.code >= 200 || message.code < 300}
            >
                {formMessage ? (
                    <>
                        <Icon
                            name={
                                message.code >= 200 || message.code < 300
                                    ? 'check circle'
                                    : 'exclamation triangle'
                            }
                        />
                        <Message.Content>
                            <Message.Header>
                                <FormattedMessage id={`${message.type}.${message.code}`} />
                            </Message.Header>
                        </Message.Content>
                    </>
                ) : (
                    <Container>
                        <Message.Header>
                            <FormattedMessage id={`${message.type}.${message.code}`} />
                        </Message.Header>
                    </Container>
                )}
            </Message>
        );
    }
    return null;
}

export default MessageModule;

MessageModule.propTypes = {
    formMessage: PropTypes.bool,
    message: PropTypes.object.isRequired,
};

MessageModule.defaultProps = {
    formMessage: false,
};
