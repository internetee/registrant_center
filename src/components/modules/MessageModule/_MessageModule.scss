.status-message.ui {
  margin: -20px 0 0;
  padding: 40px 20px 20px;
  background: #fff;
  text-align: center;
  border: 0;
  border-radius: 0;
  .header:not(.ui) {
    color: #fff;
    font-size: 20px;
    line-height: 20px;
    font-weight: 900;
  }
  &.positive {
    color: #fff;
    background: $green;
    box-shadow: none;
  }
  &.negative {
    color: #fff;
    background: $orange;
    box-shadow: none;
  }
  @include min-screen(768px) {
    margin: -20px 0 0;
    padding: 60px 40px 40px;
    .header:not(.ui) {
      font-size: 24px;
      line-height: 25px;
    }
  }
  @include min-screen(1224px) {
    margin: -40px 0 0;
    padding: 120px 50px 80px;
    .header:not(.ui) {
      font-size: 30px;
      line-height: 30px;
    }
  }
}