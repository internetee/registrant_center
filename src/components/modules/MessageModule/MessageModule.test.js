import 'core-js';
import React from 'react';
import { mount, shallow } from 'enzyme';
import MessageModule from './MessageModule';
import Providers from '../../../__mocks__/Providers';

describe('components/modules/MessageModule', () => {
	const mockErrorMessage = {
		code: 404,
		type: 'test',
	};

	const mockSuccessMessage = {
		code: 200,
		type: 'test',
	};

	it('should render', () => {
		const cmp = shallow(
			<Providers>
				<MessageModule message={mockErrorMessage} />
			</Providers>
		);
		expect(cmp).toMatchSnapshot();
		cmp.unmount();
	});

	it('should render error message', () => {
		const cmp = mount(
			<Providers>
				<MessageModule message={mockErrorMessage} />
			</Providers>
		);
		expect(cmp.find('.message').hasClass('negative')).toBeTruthy();
		cmp.unmount();
	});

	it('should render success message', () => {
		const cmp = mount(
			<Providers>
				<MessageModule message={mockSuccessMessage} />
			</Providers>
		);
		expect(cmp.find('.message').hasClass('positive')).toBeTruthy();
		cmp.unmount();
	});
});
