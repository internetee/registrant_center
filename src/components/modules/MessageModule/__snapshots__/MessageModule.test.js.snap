// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/modules/MessageModule should render 1`] = `ShallowWrapper {}`;
